app/helpers/generic_file_helper.rb

```diff
@@ -39,7 +39,19 @@ def download_path(*args)
       gf = GenericFile.find(item.id)
       return gf.doi_url if gf.doi_url.present?
     end
-    return sufia.download_path(*args)
+    if args[1].nil?
+      return "/files/" + item.id + "/" + URI::encode(item.label) unless item.label.nil?
+      return "/files/" + item.id + "/nofile"
```

Comment:
Yes, the specs were next on my list... The "nofile" is there to handle the
case of an item that doesn't have a file (which shouldn't happen in
production, but just in case). Without it, a download link will be
generated that points to the item's show view, which would be useless to
harvesters; with it, the download link leads to a 404, which is at least
appropriate to the situation and matches Sufia's default behaviour (where
/downloads/<noid> produces a 404 if there's no file attached).


Peter Binkley, Ph.D., MLIS / Digital Initiatives Technology Librarian /
peter.binkley@ualberta.ca
2-10K Cameron Library / University of Alberta / Edmonton, Alberta / Canada
T6G 2J8
phone 780-492-3743 / fax 780-492-9243

On 14 June 2016 at 16:16, pgwillia <notifications@github.com> wrote:

> In app/helpers/generic_file_helper.rb
> <https://github.com/ualbertalib/HydraNorth/pull/1191#discussion_r67068036>
> :
>
> > @@ -39,7 +39,19 @@ def download_path(*args)
> >        gf = GenericFile.find(item.id)
> >        return gf.doi_url if gf.doi_url.present?
> >      end
> > -    return sufia.download_path(*args)
> > +    if args[1].nil?
> > +      return "/files/" + item.id + "/" + URI::encode(item.label) unless item.label.nil?
> > +      return "/files/" + item.id + "/nofile"
>
> These would be good conditions to write specs for. Why is the 'nofile'
> important?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ualbertalib/HydraNorth/pull/1191/files/2c1f881dd14aad385eb2760cb8c5de21c794718a#r67068036>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAQpsN12EPy0Y4SWdGviphPFUEC-0gaWks5qLyhbgaJpZM4Iyegz>
> .
>